sgx-types/src/secs.rs

```diff
@@ -67,6 +67,15 @@ impl Spec {
     }
 }
 
+impl Default for Spec {
+    fn default() -> Self {
+        Self {
+            enc_size: unsafe { Spec::max_enc_size().unwrap() },
```

Comment:
So it appears that the general practice within the Rust community is to unconditionally issue the `CPUID` instruction. So rather than an `unsafe` block here, let's just make `max_enc_size()` safe. However, please make sure there is a safety section in the documentation for that function that clearly outlines why we made that choice.

Reply:
I'm not sure how they are doing that. 
```
error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> src/secs.rs:52:19
     |
52 |         let res = core::arch::x86_64::__get_cpuid_max(LEAF_MAX_PARAM);
     |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
     |

```

Reply:
They immediately wrap that in an unsafe block.

Reply:
It is probably not correct. But it is common practice. Which is why I suggested documenting it.